src/core/manager.c

```diff
@@ -3159,6 +3159,31 @@ static bool manager_get_show_status(Manager *m, StatusType type) {
         return false;
 }
 
+const char *manager_get_confirm_spawn(Manager *m) {
+        static bool warned = false;
+        const char *vc = m->confirm_spawn;
+        _cleanup_close_ int fd = -1;
+
+        if (!vc || path_equal(vc, "/dev/console"))
+                return vc;
+
+        /* TTYs are devices that can come and go any time, and
+         * frequently aren't available yet during early boot (consider
+         * a USB rs232 dongle...). If for any reason the configured
+         * console is not ready, we fallback to the default
+         * console. */
+        fd = open_terminal(vc, O_RDWR|O_NOCTTY|O_CLOEXEC);
+        if (fd < 0) {
+                if (!warned) {
+                        warned = true;
+                        log_warning_errno(fd, "Failed to open %s: %m, using default console", vc);
+                }
+                return "/dev/console";
+        }
+        warned = false;
+        return vc;
```

Comment:
uh, oh, this is racy.... what happens if the device disappears right after checking this, but before you then manage to open the device for good?

Also, note that closing a tty triggers a hangup (which is slow!), please do not do that needlessly.

Instead, please add the fallback code right where you need it and use the fd it returns for your work instead of throwing it away. If need wrap add open_terminal_fallback_to_console() or so that does all this properly